File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_17_mediator/java/_01_before/Restaurant.java
package com.example._03_behavioral_patterns._17_mediator.java._01_before;

public class Restaurant {

    private final CleaningService cleaningService = new CleaningService();

    public void dinner(Guest guest) {
        System.out.println("dinner " + guest);
    }

    public void clean() {
        cleaningService.clean(this);
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_17_mediator/kt/_01_before/Restaurant.kt
package com.example._03_behavioral_patterns._17_mediator.kt._01_before

class Restaurant(
    private val cleaningService: CleaningService = CleaningService()
) {
    fun dinner(guest: Guest) {
        println("dinner $guest")
    }

    fun clean() {
        cleaningService.clean(this)
    }
}
